connexion/decorators/produces.py

```diff
@@ -124,7 +124,7 @@ def wrapper(*args, **kwargs):
                 logger.debug('Endpoint returned an empty response (204)', extra={'url': url, 'mimetype': self.mimetype})
                 return '', 204, headers
 
-            data = json.dumps(data, indent=2, cls=JSONEncoder)
+            data = json.dumps(data, indent=2)
```

Comment:
This `json.dumps` come from `flask` namespace (http://flask.pocoo.org/docs/0.10/api/#flask.json.dumps) which changes the behaviour to use as encoder the class referenced by the flask app attribute `json_encoder`. That is the expected behaviour while using JSON encoders with flask. So you can personalize the encoder changing the `app.json_encoder`.

References:
http://flask.pocoo.org/snippets/119/
http://flask.pocoo.org/docs/0.10/api/#flask.Flask.json_encoder